server/usr/local/bin/add-user

```diff
@@ -0,0 +1,115 @@
+#!/bin/ash
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+# shellcheck shell=dash disable=SC3001,SC3003,SC3028
+
+usage_error()
+{
+    exec 1>&2
+    [ -n "$1" ] && printf "Error: %s\n" "$1"
+    printf "Usage: %s [OPTIONS] USER\n" "$0"
+    printf "    USER                    Add/Create a new USER account\n"
+    printf "    OPTIONS\n"
+    printf "      -g group1,group2,...  Comma-separated list of groups for initial membership \n"
+    printf "      -d display_name       The user display name\n"
+    printf "      -p password           Initial password; \"-\" reads from standard input\n"
+    printf "      -G numgid             Set the primary group number\n"
+    printf "      -U numuid             Set the numeric user identifier\n"
+    exit 2
+}
+
+ngid=1001
+# XXX default nuid? e.g. max uidNumber + 1
+nuid=$(( 1003 + RANDOM % 8996 ))
```

Comment:
Why use a random UID? Maybe it's better to get current maximum UID and add 1.

Reply:
See https://github.com/NethServer/ns8-openldap/pull/3#discussion_r866559687